File: local-movie-app/src/main/java/com/github/rahmnathan/localmovies/app/google/cast/config/CastOptionsProvider.java
package com.github.rahmnathan.localmovies.app.google.cast.config;

import android.content.Context;

import com.google.android.gms.cast.CastMediaControlIntent;
import com.google.android.gms.cast.framework.CastOptions;
import com.google.android.gms.cast.framework.OptionsProvider;
import com.google.android.gms.cast.framework.SessionProvider;
import com.google.android.gms.cast.framework.media.CastMediaOptions;
import com.google.android.gms.cast.framework.media.NotificationOptions;

import java.util.List;

public class CastOptionsProvider implements OptionsProvider {
    private static final String LOCALMOVIES_RECEIVER_APP_ID = "5F217DDB";

    @Override
    public CastOptions getCastOptions(Context appContext) {
        NotificationOptions notificationOptions = new NotificationOptions.Builder()
                .setTargetActivityClassName(ExpandedControlActivity.class.getName())
                .build();

        CastMediaOptions mediaOptions = new CastMediaOptions.Builder()
                .setNotificationOptions(notificationOptions)
                .setExpandedControllerActivityClassName(ExpandedControlActivity.class.getName())
                .build();

        return new CastOptions.Builder()
                .setReceiverApplicationId(LOCALMOVIES_RECEIVER_APP_ID)
                .setCastMediaOptions(mediaOptions)
                .build();
    }

    @Override
    public List<SessionProvider> getAdditionalSessionProviders(Context context) {
        return null;
    }
}

File: local-movie-app/src/main/java/com/github/rahmnathan/localmovies/app/google/cast/config/CastOptionsProvider.kt
package com.github.rahmnathan.localmovies.app.google.cast.config

import android.content.Context
import com.google.android.gms.cast.framework.CastOptions
import com.google.android.gms.cast.framework.OptionsProvider
import com.google.android.gms.cast.framework.SessionProvider
import com.google.android.gms.cast.framework.media.CastMediaOptions
import com.google.android.gms.cast.framework.media.NotificationOptions

class CastOptionsProvider : OptionsProvider {
    override fun getCastOptions(appContext: Context): CastOptions {
        val notificationOptions = NotificationOptions.Builder()
                .setTargetActivityClassName(ExpandedControlActivity::class.java.name)
                .build()
        val mediaOptions = CastMediaOptions.Builder()
                .setNotificationOptions(notificationOptions)
                .setExpandedControllerActivityClassName(ExpandedControlActivity::class.java.name)
                .build()
        return CastOptions.Builder()
                .setReceiverApplicationId(LOCALMOVIES_RECEIVER_APP_ID)
                .setCastMediaOptions(mediaOptions)
                .build()
    }

    override fun getAdditionalSessionProviders(context: Context): List<SessionProvider> {
        return ArrayList()
    }

    companion object {
        private const val LOCALMOVIES_RECEIVER_APP_ID = "5F217DDB"
    }
}